docs/cpu_templates/cpu-template-helper.md

```diff
@@ -0,0 +1,239 @@
+CPU template for this purpose, it is efficient to focus on the differences
+in guest CPU configurations across those CPU models. Given that a dumped guest
+CPU configuration typically amounts to approximately 1,000 lines, this command
+considerably narrows down the scope to consider.
+
+#### Verify command
+
+This command verifies that the given custom CPU template is applied correctly.
+
+```
+cpu-template-helper template verify \
+    --config <firecracker-config>
+```
+
+Firecracker modifies the guest CPU configuration after the CPU template is
+applied. Occasionally, due to hardware and/or software limitations, KVM might
+not set the given configuration. Since Firecracker does not check them at
+runtime, it is required to ensure that these situations don't happen with their
+custom CPU templates before deploying it.
+
+The command uses the same configuration file as Firecracker and the path to the
+custom CPU template file should be specified in the "cpu-config" field.
+
+> **Note**
+This command does not ensure that the contents of the template are sensible.
+Thus, users need to make sure that the template does not have any inconsistent
+entries and does not crash guests.
+
+### Fingerprint-related commands
+
+#### Dump command
+
+This command not only dumps the guest CPU configuration, but also host
+information that could affect the validity of custom CPU templates.
+
+```
+cpu-template-helper fingerprint dump \
+    --config <firecracker-config> \
+    --output <output-path>
+```
+
+Keeping the underlying hardware and software stack updated is essential for
+maintaining security and leveraging new technologies. On the other hand, since
+the guest CPU configuration can vary depending on the infrastructure, updating
+it could lead to a situation where a custom CPU template loses its validity.
+In addition, even if values of the guest CPU configuration don't change, its
+internal behavior or semantics could still change. For instance, a kernel
+version update may introduce changes to KVM emulation and a microcode update may
+alter the behavior of CPU instructions.
+
+To ensure awareness of these changes, it is strongly recommended to store the
+fingerprint file at the time of creating a custom CPU template and to
+continuously compare it with the current one.
+
+#### Compare command
+
+This command compares two fingerprint files: one was taken at the time of custom
+CPU template creation and the other is taken currently.
+
+```
+cpu-template-helper fingerprint compare \
+    --prev <prev-fingerprint> \
+    --curr <curr-fingerprint> \
+    --filters <field-1> [..<field-N>]
+```
+
+By continously comparing fingerprint files, users can ensure they are aware of
+any changes that could require revising the custom CPU template. However, it is
+worth noting that not all of these changes necessarily require a revision, and
+some detected changes could be false alarms. To provide users with flexibility
```

Comment:
nitpick: Silly thought here, but "alarm" here is not how I would view the fingerprint tool's purpose. I would have considered phrasing "false alarms" to "inconsequential to the microvm's security or performance".